TheraLang.DAL/Repository/IRepository.cs

```diff
@@ -1,23 +1,24 @@
+using System;
 using System.Collections.Generic;
 using System.Linq;
+using System.Linq.Expressions;
 using System.Threading.Tasks;
 
 namespace TheraLang.DAL.Repository
 {
     public interface IRepository<TEntity> where TEntity : class
     {
-        IQueryable<TEntity> Get();
-
-        void Remove(TEntity entity);
-       
-        void RemoveRange(IEnumerable<TEntity> entities);
-       
-        Task Add(TEntity entity);
-        
-        Task AddRange(IEnumerable<TEntity> entity);
-       
-        void Update(TEntity entity);
-      
-        void Attach(TEntity entity);
+        Task<TEntity> GetAsync(Expression<Func<TEntity, bool>> where);
```

Comment:
Do we actually need these new methods in a repository?

Reply:
Since now we follow new code style where all methods in repository is async, so its more crearly use "where" in repository with Get() together, not in servises.